File: java/src/main/java/benchmarks/SortBenchmark.java
package benchmarks;

public abstract class SortBenchmark extends Benchmark {
    protected int[] data;
    protected long sizeVal;
    private long resultVal;

    public SortBenchmark() {
        resultVal = 0L;
        sizeVal = 0L;
    }

    @Override
    public void prepare() {
        if (sizeVal == 0) {
            sizeVal = configVal("size");
            data = new int[(int) sizeVal];
            for (int i = 0; i < data.length; i++) {
                data[i] = Helper.nextInt(1_000_000);
            }
        }
    }

    abstract int[] test();

    @Override
    public void run(int iterationId) {

        resultVal += data[Helper.nextInt(data.length)];
        int[] t = test();
        resultVal += t[Helper.nextInt(data.length)];
    }

    @Override
    public long checksum() {
        return resultVal;
    }
}

File: kotlin/src/main/kotlin/benchmarks/SortBenchmark.kt
package benchmarks

import Benchmark

abstract class SortBenchmark : Benchmark() {
    protected val sizeVal = configInt("size")
    protected lateinit var data: IntArray
    protected var resultVal: UInt = 0u

    override fun prepare() {
        data = IntArray(sizeVal) { Helper.nextInt(1_000_000) }
    }

    abstract fun test(): IntArray

    override fun run(iterationId: Int) {
        resultVal += data[Helper.nextInt(sizeVal)].toUInt()
        val t = test()
        resultVal += t[Helper.nextInt(sizeVal)].toUInt()
    }

    override fun checksum(): UInt = resultVal
}
